packages/rocketchat-ui/views/app/room.coffee

```diff
@@ -418,6 +418,14 @@ Template.room.events
 		ChatMessage.update {_id: this._arguments[1]._id, 'urls.url': $(event.currentTarget).data('url')}, {$set: {'urls.$.downloadImages': true}}
 		ChatMessage.update {_id: this._arguments[1]._id, 'attachments.image_url': $(event.currentTarget).data('url')}, {$set: {'attachments.$.downloadImages': true}}
 
+	'click .collapse-switch': (e) ->
+		url = $(e.currentTarget).data('url')
+		collapsed = $(e.currentTarget).data('collapsed')
+		id = @_arguments[1]._id
+		ChatMessage.update {_id: id, 'urls.url': url}, {$set: {'urls.$.collapsed': !collapsed}}
```

Comment:
Check if url is undefined before calling this update (re: https://github.com/RocketChat/Rocket.Chat/pull/2763#issuecomment-205809574)